pykit/transport/http/HttpConnection.py

```diff
@@ -0,0 +1,63 @@
+import requests
+from ...Toolkit import Toolkit
+
+
+class HttpConnection:
+    """
+    Represents a REST HTTP connection that can be used to get the Python
+    Toolkit object.
+    """
+
+    def __init__(self, db2sock_rest_url, db2sock_auth):
+        self.db2sock_auth = db2sock_auth
+        self.db2sock_rest_url = db2sock_rest_url
+        self.payload = []
+        self.test_db2sock()
+
+    def toolkit(self):
+        """
+        Return an instance of the toolkit with a connection defined.
+
+        :return: Toolkit
+        """
+        return Toolkit(self)
+
+    def add(self, o):
+        """
+        Add an object to the payload that will be passed to DB2Sock.
+
+        :param o: Object to be added
+        :return: void
+        """
+        if isinstance(o, dict):
+            self.payload.append(o)
+        else:
+            raise TypeError('Only dictionaries are supported right now.')
+
+    def execute(self):
+        """
+        Execute the payload and then clear the payload.
+
+        :return: Response
+        """
+        response = requests.post(
+            self.db2sock_rest_url, json=self.payload, auth=self.db2sock_auth)
+        self.payload = []
+        return response
```

Comment:
Why are we leaking the response object back to the user? This will mean that when we add a new transport we would need to duplicate a requests' response object for that transport or the user would have to code their application differently based on the type of transport used. Instead, we should define our own response format.

Building on my other comment about toolkit object vs connection object, I think just passing back the JSON data to the toolkit object (which parses the data in to python objects) while if we get an error actually calling db2sock (http error, db2 error, SSH error or whatever) we could raise some sort of `TransportError`.